### PR TITLE
Optionally create an availability set

### PR DESCRIPTION
**Pull request by DatsloJRel**
## Describe your changes
Add variable to optionally create an availability set

## Checklist before requesting a review
- [X] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [X] I have executed pre-commit on my machine
- [X] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -88,6 +88,12 @@ variable "enable_accelerated_networking" {
   default     = false
 }
 
+variable "enable_availability_set" {
```
Comment (jiaweitao001) [inline variables.tf:91]: According to our new codex, I suggest renaming this variable to `availability_set_enabled`.

File: variables.tf [hunk 2]
```diff
@@ -88,6 +88,12 @@ variable "enable_accelerated_networking" {
...
+  type        = bool
+  description = "(Optional) Enable or Disable availability set.  Default is true (enabled)."
+  default     = true
```
Comment (jiaweitao001) [inline variables.tf:94]: To ensure this variable is not `null`. I suggest we would better add `nullable = false` here.

File: main.tf [hunk 3]
```diff
@@ -185,7 +185,7 @@ resource "azurerm_virtual_machine" "vm_windows" {
   network_interface_ids         = [element(azurerm_network_interface.vm[*].id, count.index)]
   resource_group_name           = var.resource_group_name
   vm_size                       = var.vm_size
-  availability_set_id           = var.zone == null ? azurerm_availability_set.vm[0].id : null
+  availability_set_id           = (var.enable_availability_set && (var.zone == null)) ? azurerm_availability_set.vm[0].id : null
```
Comment (jiaweitao001) [inline main.tf:188]: Same.

File: main.tf [hunk 4]
```diff
@@ -56,7 +56,7 @@ resource "azurerm_virtual_machine" "vm_linux" {
   network_interface_ids            = [element(azurerm_network_interface.vm[*].id, count.index)]
   resource_group_name              = var.resource_group_name
   vm_size                          = var.vm_size
-  availability_set_id              = var.zone == null ? azurerm_availability_set.vm[0].id : null
+  availability_set_id              = (var.enable_availability_set && (var.zone == null)) ? azurerm_availability_set.vm[0].id : null
```
Comment (jiaweitao001) [inline main.tf:59]: Maybe this looks simpler?

```
availability_set_id = try(azurerm_availability_set.vm[0].id, null)
```

**Review by jiaweitao001 — COMMENTED** [review]
Thank for the PR @DatsloJRel ! Overall looks good. Just as few comments to address.

File: variables.tf (hunk 1 above)
Comment (DatsloJRel) [reply variables.tf]: Agreed.  Changed in latest commit.

File: variables.tf [hunk 5]
```diff
@@ -40,6 +40,13 @@ variable "as_platform_update_domain_count" {
   default     = 2
 }
 
+variable "availability_set_enabled" {
+  description = "(Optional) Enable or Disable availability set.  Default is true (enabled)."
```
Comment (lonegunmanb) [inline variables.tf:44]: Would the following description be better?:

```hcl
description = "(Optional) Enable or Disable availability set.  Default is `true` (enabled)."
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @DatsloJRel for the update, only one tiny description issue remains.

**Review by DatsloJRel — COMMENTED** [review]
I have made the requested change.

File: variables.tf (hunk 5 above)
Comment (DatsloJRel) [reply variables.tf]: I have made the suggested change.

**Review by lonegunmanb — APPROVED** [review]
Thanks @DatsloJRel, LGTM! 🚀

**Pull request merged**